### PR TITLE
[TK] Add test for fused attention and ops needed

### PR DESCRIPTION
**Pull request by Groverkss**
This patch adds a test for f16 attention. The fused_attention kernel needed dtype conversion so that is also added. As a side effect, implicit vector conversion (for unary ops) is now also supported.

### CODE REVIEW

File: core/shark_turbine/kernel/compiler/builder.py [hunk 1]
```diff
@@ -217,10 +238,33 @@ def unary_vector_arithmetic(self, op: str, val: IRProxyValue) -> IRProxyValue:
...
+        # Check bitwidth to determine if we need to extend or narrow
+        from_type = value.type
+        from_bitwidth = self.get_float_bitwidth(from_type)
+        to_bitwidth = self.get_float_bitwidth(to_type)
+        if from_bitwidth < to_bitwidth:
+            return arith_d.extf(to_type, value)
+        elif from_bitwidth > to_bitwidth:
+            return arith_d.truncf(to_type, value)
+        else:
+            raise CodegenError(f"NYI: Cast from {from_type} to {to_type}")
```
Comment (Groverkss) [inline core/shark_turbine/kernel/compiler/builder.py:267]: I don't know how to convert between bf16 and fp16

Comment (stellaraccident) [reply core/shark_turbine/kernel/compiler/builder.py:267]: No one does :) bf16 was made in protest of fp16...

Best that can be done is cast to f32 then to f16. The former is just bit twiddling.

Comment (stellaraccident) [reply core/shark_turbine/kernel/compiler/builder.py:267]: Or leave it unimplemented with error.

**Review by stellaraccident — APPROVED** [review]

File: core/shark_turbine/kernel/compiler/builder.py [hunk 2]
```diff
@@ -87,28 +87,53 @@ def __init__(
 
 class _ScalarBuilder:
     def is_floating_point_type(self, t: IrType) -> bool:
-        return str(t) in FLOAT_TYPES_ASM
+        return str(t) in FLOAT_BITWIDTHS
```
[the remark this answers is not in the document]
Comment (Groverkss) [reply core/shark_turbine/kernel/compiler/builder.py:91]: oh, cool! Added a TODO, will start using once we bump

File: core/shark_turbine/kernel/compiler/builder.py (hunk 1 above)
Comment (Groverkss) [reply core/shark_turbine/kernel/compiler/builder.py:268]: welp, leaving it unimplemented then

**Pull request merged**